docs/wiki/protocol/data-structures.md

```diff
@@ -0,0 +1,59 @@
+# Data Structures in Ethereum
+
+<br />
+
+## Primer on Merkle Tree
+
+Merkle tree is a hash-based data structure which is very effecient at data integrity and verification. It is a tree based structure where the leaf nodes hold the data values and each non-leaf node is a hash of its child nodes.
```

Comment:
This is first intro to merkle trees so you can elaborate a bit more on it's properties and features. You should also link resources to hashes, merkle trees and tree structures